src/builder/QuarkBuilder.sol

```diff
@@ -93,42 +96,80 @@ contract QuarkBuilder {
-                // ^^^ look up USDC price feeds for each supported chain?
-                // we only need USDC/USD and only on chains 1 (mainnet) and 8453 (base mainnet).
-            } else {
-                quarkOperations[actionIndex++] = Actions.bridgeUSDC(
-                    Actions.BridgeUSDC({
-                        chainAccountsList: chainAccountsList,
-                        assetSymbol: transferIntent.assetSymbol,
-                        amount: transferIntent.amount,
-                        // where it comes from
-                        originChainId: 8453, // FIXME: originChainId
-                        sender: address(0), // FIXME: sender
-                        // where it goes
-                        destinationChainId: transferIntent.chainId,
-                        recipient: transferIntent.recipient,
-                        blockTimestamp: transferIntent.blockTimestamp
-                    })
-                );
-                // TODO: also append a Actions.Action to the actions array.
-                // See: BridgeUSDC TODO for returning a Actions.Action.
+            // Note: Assumes that the asset uses the same # of decimals on each chain
+            uint256 balanceOnDstChain =
+                Accounts.getBalanceOnChain(transferIntent.assetSymbol, transferIntent.chainId, chainAccountsList);
+            uint256 amountLeftToBridge = transferIntent.amount - balanceOnDstChain;
+
+            uint256 bridgeActionCount = 0;
+            // TODO: bridge routing logic (which bridge to prioritize, how many bridges?)
+            // Iterate chainAccountList and find upto 2 chains that can provide enough fund
+            // Backend can provide optimal routes by adjust the order in chainAccountList.
+            for (uint256 i = 0; i < chainAccountsList.length; ++i) {
+                if (amountLeftToBridge == 0) {
+                    break;
+                }
+
+                Accounts.ChainAccounts memory srcChainAccounts = chainAccountsList[i];
+                if (srcChainAccounts.chainId == transferIntent.chainId) {
+                    continue;
+                }
+
+                if (
+                    !BridgeRoutes.canBridge(srcChainAccounts.chainId, transferIntent.chainId, transferIntent.assetSymbol)
+                ) {
+                    continue;
+                }
+
+                Accounts.AssetPositions memory srcAssetPositions =
+                    Accounts.findAssetPositions(transferIntent.assetSymbol, srcChainAccounts.assetPositionsList);
+                Accounts.AccountBalance[] memory srcAccountBalances = srcAssetPositions.accountBalances;
+                // TODO: Make logic smarter. Currently, this uses a greedy algorithm.
+                for (uint256 j = 0; j < srcAccountBalances.length; ++j) {
+                    // TODO: May need to factor in bridge fees for non-CCTP bridges
+                    if (srcAccountBalances[j].balance < BRIDGE_MINIMUM_AMOUNT_USDC) {
```

Comment:
I'm not against minimums, but this is weird since this constant is in USDC but we're... not checking the asset as USDC? So we'd basically need one parameter per bridgeable-asset? Plus this means we might get weird behavior like I want to send $1 over the bridge and it says "no, you can't"

Reply:
I think the better approach, generally, would be to try and consolidate. As in, if it takes two bridge actions to accomplish what one can do, always do the one first. One way to represent this would be to (for instance) try this without allowing split bridging, and then try it again if that's insufficient _with_ split bridging.

Reply:
Correct, we would ultimately need a config of minimums for each bridgeable asset. I have removed the minimum for now to simplify the logic a bit and give us room to explore alternatives (e.g. the one you suggested). I added a TODO for this.